File: atcoder_beginners_selection/ABC081B.rb
n = gets.to_i
a = gets.chomp.split(" ").map(&:to_i)
ck = true
cnt = 0
while  ck == true do
  i = 0
  n.times do
	if a[i] % 2 == 1 then
		ck = false
	else
		a[i] = a[i] / 2
		i += 1
	end
  end
  cnt += 1 if ck == true
end

print cnt

File: atcoder_beginners_selection/abc081b.py
n = int(input())
arr = list(map(int,input().split()))
cnt = 0
flg = True
while True:
  for i, value in enumerate(arr):
    if value % 2 == 0:
       arr[i] = int(value / 2)
    else:
      flg = False
      break
  if flg == False:
    break
  else:
    cnt += 1
print(cnt)